hot-reload-analysis/src/main/kotlin/org/jetbrains/compose/reload/analysis/resolveDirty.kt

```diff
@@ -49,7 +49,16 @@ private fun RuntimeInfo.resolveDirtyMethods(redefined: RuntimeInfo): List<Method
     return redefined.methodIndex.mapNotNull { (methodId, redefinedMethod) ->
         val previousMethod = methodIndex[methodId] ?: return@mapNotNull redefinedMethod
         if (previousMethod.rootScope.hash != redefinedMethod.rootScope.hash) {
-            return@mapNotNull redefinedMethod
+            if (redefinedMethod.rootScope.isComposeEntry) {
```

Comment:
I think the exception works pretty well here. 
What do you think about trying to formalize the concept as 'validation' (or any other name) and giving it its own space? If we keep the code here maybe we should return a `Try` instead of throwing? Or an `Either` that models the rejection cases.

Reply:
Yes, I think extracting the validation as a separate step during the reload is a very good idea. I will work on that